sql/src/main/java/org/apache/druid/sql/calcite/planner/SqlResourceCollectorShuttle.java

```diff
@@ -69,10 +68,9 @@ public SqlNode visit(SqlIdentifier id)
         // 'schema'.'identifier'
         if (qualifiedNameParts.size() == 2) {
           final String schema = qualifiedNameParts.get(0);
-          if (druidSchemaName.equals(schema)) {
-            resources.add(new Resource(qualifiedNameParts.get(1), ResourceType.DATASOURCE));
-          } else if (NamedViewSchema.NAME.equals(schema)) {
-            resources.add(new Resource(qualifiedNameParts.get(1), ResourceType.VIEW));
+          final String resourceType = plannerContext.getSchemaResourceType(schema);
+          if (resourceType != null) {
```

Comment:
Before it seemed that only DATASOURCE and VIEW being mapped to resources. How do we ensure this old behavior if desired?

Reply:
Oh I see, the default implementation for the other schema types will return null resource type. nvm, all good.